### DISCUSSION TITLE
Can LosslessCut detect duplicate segments between videos?

### DISCUSSION THREAD

**Discussion by dezii98** [mifi/lossless-cut#1484] [Q&A] [answered]
I have:

- Video A that record a stream from 0:00 to 3:30
- Video B that record also that stream from 3:00 to 5:00

Before, I opened 2 LosslessCut, loaded both videos and removed the last 30s from A then merge videos manually.
But with the recent 3.50.0 version, it seems I cannot open 2 LosslessCut at the same time anymore.
Does LosslessCut have any feature that auto detect duplicate segments between videos and remove them when merging?

Thank you!

**Comment by mifi** [comment, marked as answer]
see https://github.com/mifi/lossless-cut/blob/master/cli.md#multiple-instances-experimental

> Does LosslessCut have any feature that auto detect duplicate segments between videos and remove them when merging?

no such feature

**Reply by dezii98** [reply]
Thank you for your quick respone
But since I'm not very familiar with Command line interface, I have bought the window store version.
Can you explain how to open 2 LC with the window store version please?

Thank you!

**Reply by mifi** [reply]
oh I see! I had just assumed that it was never possible to open 2 instances of losslesscut from the Windows Store. (not sure why, but I assumed that Windows had some built in protection agains opening apps twice), but clearly I was wrong.

Can you find your path to losslesscut executable? I **think** it should be`C:\Program Files\WindowsApps\57275mifi.no.LosslessCut_3.50.0.0_x64__m3evqk1w5z09c\LosslessCut.exe` but please double check that.

Then you can open a terminal and run:

```bash
C:\Program Files\WindowsApps\57275mifi.no.LosslessCut_3.50.0.0_x64__m3evqk1w5z09c\LosslessCut.exe --allow-multiple-instances
```

Let me know if that works. If not I'm going to have to do a hotfix to make it possible to configure this thru settings.

**Reply by dezii98** [reply]
Thank you I did it!

The path to LosslessCut executable was: `C:\Program Files\WindowsApps\57275mifi.no.LosslessCut_3.50.0.0_x64__eg8x93dt4dxje\app\LosslessCut.exe`

I have create a shortcut with `--allow-multiple-instances` command in target box then it works like a charm

The only thing that it seems this version is different from the one I used to open from start menu (this has 2 sunflowers at start screen)

But thank you!

**Reply by mifi** [reply]
That is very interesting. That means the app is no longer correctly identifying itself as a Windows Store build, because the sunflower is only supposed to show up on non-windows store. There will be some other quirks too, like the app will check and notify about updates. (which it's not supposed to do because windows store handles that.)

Some technical notes for myself (feel free to ignore):
- LosslessCut detects itself as a Windows Store version if `process.windowsStore` is true.
- `process.windowsStore` is true if the function [GetPackageFamilyName](https://learn.microsoft.com/en-us/windows/win32/api/appmodel/nf-appmodel-getpackagefamilyname) in the Windows API returns `ERROR_SUCCESS`. ([relevant Electron code](https://github.com/electron/electron/blob/fcfbcbc7e1e9f0cf92df8b66463bf5ed299503b0/shell/common/application_info_win.cc#L77)
- Normally this is the case for people running the Windows Store version, but it seems that when you're running LosslessCut.exe from the command line, this function will probably return `APPMODEL_ERROR_NO_PACKAGE` https://stackoverflow.com/questions/39609643/determine-if-c-application-is-running-as-a-uwp-app-in-desktop-bridge-project
- From some googling, I think this is happening because when the app is run from the command line, it is no longer being run as a UWP application, but instead as a win32 application.

I found out that it seems to be possible to run an [UWP app also from the command line](https://www.addictivetips.com/windows-tips/open-uwp-apps-from-command-line-windows-10/), probably something like this:

```
explorer.exe shell:appsFolder\57275mifi.no.LosslessCut_3.50.0.0_x64__eg8x93dt4dxje!LosslessCut.app
```

...however I don't think it works with arguments. In order to run it with arguments, it seems to get more complicated: https://stackoverflow.com/questions/47765108/uwp-run-exe-file-with-parameters

Another problem with your approach is that when a new version comes out, your shortcut will break.

I think I will just implement a setting instead of a command line flag.
I hope your workaround is OK for now, then I will include this setting in the next release.

**Reply by mrienstra** [reply]
Docs link, permalink versions (locked to particular commit):
Older: [1b2291a cli.md#multiple-instances-experimental](https://github.com/mifi/lossless-cut/blob/1b2291a/cli.md#multiple-instances-experimental)
Newer: [ee29958 issues.md#multiple-losslesscut-instances](https://github.com/mifi/lossless-cut/blob/ee29958/issues.md#multiple-losslesscut-instances)
